lua/neo-tree/ui/renderer.lua

```diff
@@ -163,13 +163,15 @@ M.close = function(state, focus_prior_window)
     end
     state.winid = nil
   end
-  local bufnr = utils.get_value(state, "bufnr", 0, true)
-  state.bufnr = nil
-  vim.schedule(function()
-    if bufnr > 0 and vim.api.nvim_buf_is_valid(bufnr) then
-      vim.api.nvim_buf_delete(bufnr, { force = true })
-    end
-  end)
+  if window_existed then
```

Comment:
Why does it matter if the window existed? Why should a buffer be allowed to remain if the window was closed by some other means, but not if it was closed by this function?

Also, why is it **always** scheduled even though it usually should be deleted synchronously?

I think the code from 3.22 was already correct and would fix this issue and we should just go back to that:

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8afbb06081ce1e4beb5b18945d14a608b10babeb/lua/neo-tree/ui/renderer.lua#L167-L175

Whatever the other edge case was which the changes were trying to fix should be handled another way.

Reply:
I have confirmed that the old code works correctly and the bug in #1415 was the direct result of changing that code. I'm going to close this because reverting that code is the real fix.